radix-engine-common/src/types/consensus.rs

```diff
@@ -0,0 +1,89 @@
+use sbor::Sbor;
+
+/// A type-safe consensus epoch number.
+/// Assuming one epoch per minute (i.e. much faster progression than designed), this gives us time
+/// until A.D. ~5700.
```

Comment:
Feels awfully close, and sounds like the same argument peeps had with `IPv4` and `32-bit` architecture. Is there a reason we don't just go with `u64`? I mightve missed this discussion.

Reply:
Hah, that's true... although those people didn't control the growth rate of the number of devices/networks connected to the internet globally.

Even with 1 epoch per second we have ~60 years. If our design changes to even more extreme rates (or if someone hacks the epoch progression) then we probably have more things to fix (and data migrations to perform) than `u32 -> u64` anyways.

That said: I don't push hard for this change myself. But more performance-aware people (@dhedey) argued that "it would save us a non-trivial amount of bytes in transactions" (see https://rdxworks.slack.com/archives/C01HK4QFXNY/p1684235019062509).

Reply:
I guess we could keep it as u64 and only use u32 in the V1 transaction headers. I don't feel too strongly. Happy to be overruled if anyone does - everyone seemed pro the change on slack 👍 

But yeah - now that it's a single type, it's super easy to change I guess :)

Reply:
I've now migrated to `u64` (according to https://rdxworks.slack.com/archives/C01HK4QFXNY/p1685124666394689?thread_ts=1684235019.062509&cid=C01HK4QFXNY)